testing/vector.cu

```diff
@@ -4,7 +4,9 @@
 #include <vector>
 #include <list>
 #include <limits>
-
+#if __cplusplus >= 201103L
+  #include <utility>
```

Comment:
We may as well unconditionally `#include <utility>`, no matter the C++ version, since the header will always exist.

Eliminating the `#if` guard here (and in the other files it is used) should help reduce confusion.